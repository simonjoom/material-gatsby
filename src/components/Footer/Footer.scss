@import "../../layout/theme.scss";

.footer {
  justify-content: center;
  align-content: center;
  background-color: $md-primary-color;
  padding: 10px 0px;

  .notice-container {
    flex-direction: column;
    background-color:white;
    display: flex;
    align-content: center;
    align-items: center;
    margin-top: 25px;

    .image-section-one{

      img{
        margin: 10px;
      }
    }

    .image-section-two{
      margin-bottom: 20px;

      img{
        margin: 10px;
        width: 270px;
        border: 5px solid rgb(194, 186, 186)
      }
    }
    .contact-us{
      margin: 30px 0px;
      box-shadow: 0px 0px 10px 0px black;
      width: 30%;
      justify-content: center;
      align-items: center;
      height: 100px;
      border-radius: 10px;

      .contact-item{
        padding: 5px;
      }
    }
    h4 {
      color: $md-secondary-color;
      text-align: center;
      margin: 0;
    }
  }

  .bottom-section{
    flex-direction: column;
    display: flex;
    align-content: center;
    align-items: center;
    margin-top: 25px;
  
      .rss {
        display: flex;
        @media (max-width: 640px - 1px) {
          display: none;
        }
        flex: 1;
        justify-content: center;
      }

      .based-on {
        display: flex;
        @media (max-width: 640px - 1px) {
          display: none;
        }
        flex: 1;
        justify-content: flex-end;
      }

      .copyright {
          display: flex;
          flex: 1;
          justify-content: flex-start;
          @media (max-width: 640px - 1px) {
            justify-content: center;
          }
          h4{
            font-size: 0.9em;
          }
      }
  }

  
  
}

.footer-fixed {
  position: fixed;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 15;
}