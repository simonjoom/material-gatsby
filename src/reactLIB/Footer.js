import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames'; 

class Footer extends Component {
  render() {
    const {
      children,
      className,
      copyrights,
      links,
      moreLinks,
      ...props
    } = this.props;

    let classes = {
      'page-footer': true
    };

    return (
      <footer className={cx(classes,"md-grid", className)} {...props}>
        <div className="md-grid">
          <div className="md-cell md-cell--6-phone md-cell--6 post md-cell--8-tablet">
              {children}
          </div>
          <div className="md-cell md-cell--2-desktop-offset md-cell--6-tablet-offset md-cell--4-phone md-cell--4 post md-cell--4-tablet">
              {links} 
          </div>
        </div>
        <div className="footer-copyright">
          <div className="container">
            {copyrights}
            {moreLinks}
          </div>
        </div>
      </footer>
    );
  }
}
Footer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  copyrights: PropTypes.string,
  links: PropTypes.node,
  moreLinks: PropTypes.node
};

export default Footer;
