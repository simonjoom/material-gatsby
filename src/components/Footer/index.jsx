import React, { Component } from "react";
import Button from "react-md/lib/Buttons";
import { View, Text } from "react-native";
import { Link } from "gatsby";
import UserLinks from "../UserLinks";
import config from "../../../data/SiteConfig";
import "./Footer.scss";

class Footer extends Component {
  render() {
    const url = config.siteRss;
    const { userLinks } = this.props;
    const { copyright, fixedFooter } = config;
    if (!copyright) {
      return null;
    }
    return (
      <footer className={fixedFooter ? "footer footer-fixed" : "footer"}>
        {userLinks ? <UserLinks config={config} labeled /> : null}

        <View className="notice-container">
          <Text>
            <i className="fa fa-mobile" /> Call Us: +33 6 75 505209{" "}
            <i className="fa fa-phone" />
          </Text>
          <div className="c1 c5 c2 py2 fit center mx-auto">
            <img src="https://skiscool.com/dist/baton1024.jpg" />
            <img src="https://skiscool.com/dist/tripadvisor.jpg" />
          </div>
          <div
            id="ime"
            className="flex flex-wrap justify-center relative py2 fit center mx-auto"
          >
            <img src="https://skiscool.com/dist/pic/courchevel.jpg" />
            <img src="https://skiscool.com/dist/pic/apogee.jpg" />
            <img src="https://skiscool.com/dist/pic/davidcintract.jpg" />
          </div>

          <div className="copyright">
            <h4>{copyright}</h4>
          </div>

          <div className="rss">
            <Link to={url}>
              <Button flat secondary iconClassName="fa fa-rss">
                Subscribe
              </Button>
            </Link>
          </div>

          <div className="based-on">
            <h4>
              Based on{" "}
              <a href="https://github.com/simonjoom/material-gatsby">
                Gatsby SkiScool
              </a>
              .
            </h4>
          </div>
        </View>
      </footer>
    );
  }
}

export default Footer;
