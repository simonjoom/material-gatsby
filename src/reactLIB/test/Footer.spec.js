import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../src/Footer';

let moreLinks = (
  <a className="grey-text text-lighten-4 right" href="#!">
    More Links
  </a>
);
let links = (
  <ul>
    <li>
      <a className="grey-text text-lighten-3" href="#!">
        Link 1
      </a>
    </li>
  </ul>
);
let wrapper = shallow(
  <Footer
    moreLinks={moreLinks}
    copyrights="&copy; 2015 Copyright Text"
    links={links}
  >
    <p>Footer Content</p>
  </Footer>
);

describe('<Footer />', () => {
  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
