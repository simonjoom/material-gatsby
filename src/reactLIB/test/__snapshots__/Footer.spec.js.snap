// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders 1`] = `
<footer
  className="page-footer"
>
  <div
    className="container"
  >
    <Row>
      <Col
        l={6}
        s={12}
      >
        <p>
          Footer Content
        </p>
      </Col>
      <Col
        l={4}
        offset="l2"
        s={12}
      >
        <ul>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#!"
            >
              Link 1
            </a>
          </li>
        </ul>
      </Col>
    </Row>
  </div>
  <div
    className="footer-copyright"
  >
    <div
      className="container"
    >
      Â© 2015 Copyright Text
      <a
        className="grey-text text-lighten-4 right"
        href="#!"
      >
        More Links
      </a>
    </div>
  </div>
</footer>
`;
